### ISSUE TITLE
Task parameter defaults are ignored

### ISSUE THREAD

**Issue by etiennebarrie** [Shopify/maintenance_tasks#485]
Since we have form inputs for all the task parameters, parameter defaults are overridden by form values, even when the form input is empty.

A possible workaround is to open developer tools and remove the input or disable it. 😅 

It works with the command-line runner.

In terms of fix, I think we can't know if an attribute has a default because it's stored in `_default_attributes` and it looks like a pretty internal API. We could populate the fields with their default values but that seems a bit fishy to call the default, serialize it to a string, put in the HTML, get it back through HTTP and deserialize it back to an object. I mean that could work.

Maybe instead we could just remove attributes from the hash where the value is just an empty string, and let Active Model do the default only once before enqueuing.

**Comment by nvasilevski** [comment]
>We could populate the fields with their default values but that seems a bit fishy to call the default, serialize it to a string, put in the HTML, get it back through HTTP and deserialize it back to an object.

For some reason I'm leaning towards this option. I'm terrible at UX but I feel like this seems to be a very user-friendly option, to actually see default values on the form
For example having `attribute :content, :string, default: "default content"` attribute we would see it as:
![image](https://user-images.githubusercontent.com/5512772/142268969-a4212c49-4b3b-4eef-a30a-e9db924c4145.png)

The problem that `_default_attributes` indeed seems to be a private API. Wonder if we could consider exposing default values publicly in Rails 🤔 
Should we try getting default values from the object, for example:
```ruby
irb(main):005:0> Maintenance::ParamsTask.new.attributes
=> {"post_ids"=>nil, "content"=>"default content"}
```

It seems to be more or less common way of rendering form for a non-persisted object in the scope of the `new` action for a scaffold-generated Rails model 🤔

**Comment by etiennebarrie** [comment]
Yes using a new object and accessing the value is the only API I know to get to the defaults. Serializing/deserializing to a HTML form inputs is probably OK most of the time but will be a bit weird with booleans/time etc. Maybe we could get a bit fancy on the inputs depending on the type.

**Comment by nvasilevski** [comment]
>Maybe we could get a bit fancy on the inputs depending on the type.

Yep, definitely 👍 
I think it would be reasonable to go ahead and render default values in the existing `text_area` fields and consider rendering different input types per attribute type as a separate task
